### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by ram-g-athreya**

### CODE REVIEW

File: mgnn/get_relation_emb.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import numpy as np
+import config_predicate as args
```
Comment (rrichajalota) [inline mgnn/get_relation_emb.py:2]: @GingerEater, I think the config_predicate file is missing.

Comment (GingerEater) [reply mgnn/get_relation_emb.py:2]: Hi, this file is not used in our model and I keep it just for any eventuality. I'll delete all of useless files when we finish the whole program~

Comment (GingerEater) [reply mgnn/get_relation_emb.py:2]: "this file" is pointing to get_relation_emb.py

Comment (rrichajalota) [reply mgnn/get_relation_emb.py:2]: For now, please comment out the code that you are not using and later you can remove it. :)

File: baseline/baseline.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            gram_2 = get_ngram(tmp, 2)
+            for gram in gram_2:
+                # if not g.emb(word)[0]:
+                #     sentence_emb.append(np.random.uniform(-0.01, 0.01, size=(1, 300))[0])
+                # else:
+                #     sentence_emb.append(np.array(g.emb(word)))
+                emb1 = fasttext[gram[0]].numpy(); emb2 = fasttext[gram[1]].numpy()
+                sentence_emb.append(emb1+emb2)
+            sentence_emb = np.array(sentence_emb)
+
```
Comment (ram-viv-labs) [inline baseline/baseline.py:125]: You can remove lines 91 to 99 and instead use:

`sentence_emb = get_ngram_embeddings(tmp, 2)`

File: baseline/baseline.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+from nltk.tokenize import word_tokenize
+from pretreatment.DataExtract import EntityLinking, GetPredicateList, Entity_Link_Falcon
+from pretreatment.QueryFilter import *
+from torchnlp.word_to_vector import FastText, GloVe
+fasttext = FastText()
+from embeddings import GloveEmbedding
+g = GloveEmbedding('common_crawl_840', d_emb=300)
+import math
+
+def get_ngram(text, n):
```
Comment (ram-viv-labs) [inline baseline/baseline.py:21]: Could you use this implementation instead. It returns both 1 and 2 word embeddings for the words in the sentence. For multi words the embeddings are averaged over.


```
def get_ngram_embeddings(text, n):
  embeddings = []
  for i in range(len(text)):
    for j in range(1, n + 1):
      words = text[i: i + j]
      if len(words > 0):
        embedding = [] # Please test that this line works as expected I was not able to exhaustively run this code
        for word in words:     
          embedding += fasttext[word].numpy()
        embeddings.append(embedding / len(words))
  return embeddings
```

File: baseline/baseline.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+                if predicate in idf:
+                    idf_score = idf[predicate]
+                else:
+                    idf_score = 2.0
+
+                # if not g.emb(predicate)[0]:
+                #     predicate_emb = np.array([np.random.uniform(-0.01, 0.01, size=(1, 300))[0]])
+                # else:
+                #     predicate_emb = np.array([g.emb(predicate)])
+                predicate_emb = np.array([fasttext[predicate].numpy()])
```
Comment (ram-viv-labs) [inline baseline/baseline.py]: Please do word tokenization here. Some predicates are multi word. Calculate embedding for each word and then take an average.

`Eg: <http://dbpedia.org/ontology/routeEnd> => 'route end' which is the 3rd question in the results`

**Review by ram-viv-labs — CHANGES_REQUESTED** [review]

File: baseline/baseline.py (hunk 3 above)
Comment (GingerEater) [reply baseline/baseline.py:21]: Got it! I'll replace that

**Pull request merged**